### PR TITLE
feat(test): Add 1ft impl. E2E tests (cypress.io)

### PR DESCRIPTION
**Pull request by jdelpierre**
## Description

For a necessity of test automation on the UX/UI Front-End side, the "End To End" part is initiated with the Cypress.io framework.

Refs: MON-6329

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/end_to_end/cypress/integration/Login.feature [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Feature: The Centreon homepage
+
+  I want to open centreon homepage
+
+  @focus
+  Scenario: login with valid credentials
+    Given I am on the login page
+    When I fill in inputs credentials files user data
```
Comment (bdauria) [inline tests/end_to_end/cypress/integration/Login.feature:8]:
```suggestion
    When I type the user credentials
```

File: tests/end_to_end/cypress/integration/Login.feature [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    And I press "Connect"
+    Then I should see "Header content"
```
Comment (bdauria) [inline tests/end_to_end/cypress/integration/Login.feature:10]:
```suggestion
    Then I should see the Header
```

File: tests/end_to_end/cypress/integration/Login/login.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Given, When, And, Then } from "cypress-cucumber-preprocessor/steps";
+
+Given('I am on the login page', () => {
+  cy.visit('http://localhost:3400')
+})
+
+When('I fill in inputs credentials files user data', () => {
```
Comment (bdauria) [inline tests/end_to_end/cypress/integration/Login/login.ts:7]:
```suggestion
When('I type the user credentials', () => {
```

File: tests/end_to_end/cypress/integration/Login/login.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      cy.get('input[name=useralias]').type(user.login)
+      cy.get('input[name=password]').type(user.password)
+    })
+})
+
+And('I press "Connect"', () => {
+  cy.get('form').submit()
+})
+
+Then('I should see "Header content"', () => {
```
Comment (bdauria) [inline tests/end_to_end/cypress/integration/Login/login.ts:20]:
```suggestion
Then('I should see the Header', () => {
```

File: tests/end_to_end/cypress/support/index.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import './commands'
+
+beforeEach(() => {
+  cy.dockerStart();
+});
+
+afterEach(() => {
+// Added this time to delay the end of video recording
+  cy.wait(5000)
```
Comment (bdauria) [inline tests/end_to_end/cypress/support/index.ts:9]: Why?

Comment (jdelpierre) [reply tests/end_to_end/cypress/support/index.ts]: It was during the "tests", I forgot to take it off.

File: tests/end_to_end/cypress/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+This allows to manage scenarios written in .feature format (Gerkhin syntax).
+
+> https://github.com/TheBrainFamily/cypress-cucumber-preprocessor
+
+## Step by Step
+
+### Create the Centreon Docker testing environment
+
+> cypress:build:docker
```
Comment (Thebarda) [inline tests/end_to_end/cypress/README.md:30]:
```suggestion
> npm run cypress:build:docker
```

File: tests/end_to_end/cypress/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+optional parameters :
+
+1. Version centreon [default = the last of master]
+2. Image docker name [default = ‘mon-web’]
+3. distribution [default = centos7]
+
+### Start and Stop dockers testing environment
+
+> cypress:env:start or cypress:env:stop
```
Comment (Thebarda) [inline tests/end_to_end/cypress/README.md:40]:
```suggestion
> npm run cypress:env:start or npm run cypress:env:stop
```

File: tests/end_to_end/cypress/README.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+These 2 commands allow to launch the scripts to start or stop the Dockers built by the previous command
+
+## Open the Cypress dashboard or Run by CLI
+
+> cypress:open
```
Comment (Thebarda) [inline tests/end_to_end/cypress/README.md:46]:
```suggestion
> npm run cypress:open
```

File: tests/end_to_end/cypress/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+This command allows to open a Cypress UI to manipulate and execute the tests manually.
+
+> cypress:run
```
Comment (Thebarda) [inline tests/end_to_end/cypress/README.md:50]:
```suggestion
> npm run cypress:run
```

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**